src/convert/datetime.jl

```diff
@@ -6,8 +6,14 @@ rcopy(::Type{Date}, s::Ptr{RealSxp}) = rcopy(Date, s[1])
 rcopy(::Type{DateTime}, s::Ptr{RealSxp}) = rcopy(DateTime, s[1])
 
 rcopy(::Type{Date}, x::Float64) = Date(Dates.UTInstant(Dates.Day((isnan(x) ? 0 : x) + 719163)))
-rcopy(::Type{DateTime}, x::Float64) =
-    DateTime(Dates.UTInstant(Dates.Millisecond(((isnan(x) ? 0 : x) + 62135683200) * 1000)))
+function rcopy(::Type{DateTime}, x::Float64)
+    ms = ((isnan(x) ? 0 : x) + 62135683200) * 1_000
+    if !isinteger(ms)
+        @debug "Precision lost in conversion to DateTime"
+    end
+    ms = round(Int, ms)
```

Comment:
Since you're rounding anyway, you could skip the `trunc` that `isinteger` does and just check for equality of values pre- and post-`round`. Doesn't really matter though.

Reply:
yep! hence #535